tests/app/forms/test_questionnaire_form.py

```diff
@@ -16,1677 +16,1678 @@
 from app.questionnaire import QuestionnaireSchema
 from app.questionnaire.placeholder_renderer import PlaceholderRenderer
 from app.utilities.schema import load_schema_from_name
-from tests.app.app_context_test_case import AppContextTestCase
 
 
-# pylint: disable=no-member
-# wtforms Form parents are not discoverable in the 2.3.3 implementation
-class TestQuestionnaireForm(
-    AppContextTestCase
-):  # noqa: C901  pylint: disable=too-many-public-methods
-    @staticmethod
-    def _error_exists(answer_id, msg, mapped_errors):
-        error_id = f"{answer_id}-error"
-        return any(
-            e_id == error_id and str(msg) in ordered_errors
-            for e_id, ordered_errors in mapped_errors
-        )
+def _error_exists(answer_id, msg, mapped_errors):
+    error_id = f"{answer_id}-error"
+    return any(
+        e_id == error_id and str(msg) in ordered_errors
+        for e_id, ordered_errors in mapped_errors
+    )
 
-    def test_form_ids_match_block_answer_ids(self):
-        with self.app_request_context():
-            schema = load_schema_from_name("test_textfield")
 
-            question_schema = schema.get_block("name-block").get("question")
+def test_form_ids_match_block_answer_ids(app, answer_store, list_store):
+    with app.test_request_context():
+        schema = load_schema_from_name("test_textfield")
 
-            form = generate_form(
-                schema,
-                question_schema,
-                AnswerStore(),
-                ListStore(),
-                metadata=None,
-                response_metadata={},
-            )
+        question_schema = schema.get_block("name-block").get("question")
 
-            for answer_id in schema.get_answer_ids_for_block("name-block"):
-                self.assertTrue(hasattr(form, answer_id))
+        form = generate_form(
+            schema,
+            question_schema,
+            answer_store,
+            list_store,
+            metadata=None,
```

Comment:
Since you are now refactoring, might as well fix this.  It is expecting `Mapping` 
```suggestion
            metadata={},
```